nixos/modules/services/mail/public-inbox.nix

```diff
@@ -0,0 +1,569 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.public-inbox;
+  stateDir = "/var/lib/public-inbox";
+
+  manref = name: vol: "<citerefentry><refentrytitle>${name}</refentrytitle><manvolnum>${toString vol}</manvolnum></citerefentry>";
+
+  singleIniAtom = with types; nullOr (oneOf [ bool int float str ]) // {
+    description = "INI atom (null, bool, int, float or string)";
+  };
+  iniAtom = with types; coercedTo singleIniAtom singleton (listOf singleIniAtom) // {
+    description = singleIniAtom.description + " or a list of them for duplicate keys";
+  };
+  iniAttrs = with types; attrsOf (either (attrsOf iniAtom) iniAtom);
+  gitIni = {
+    type = with types; attrsOf iniAttrs;
+    generate = name: value: pkgs.writeText name (generators.toGitINI value);
+  };
```

Comment:
Why does public-inbox use the Git INI format? Do they intentionally do that or is it coincidence that they're the same?

But yeah I agree that it makes sense to add `gitIni` to `pkgs.formats` since there's already a `lib.generators` for it

Reply:
The use of Git INI is intentional: https://public-inbox.org/public-inbox-config.txt

public-inbox stores mailing list archives in a git repository, so it's tightly coupled with git.

Reply:
I've added `gitIni` to `pkgs.formats`. Some types (`iniAttrs`, `iniAtom`) are still copied into `nixos/modules/services/mail/public-inbox.nix` because they're used in `types.submodule` of the `settings`.